### PR TITLE
`General`: Add profile pictures to course user list and user administration

### PR DESCRIPTION
**Pull request by PaRangger**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data (e.g. using paging).
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

Currently the user profile pictures are shown in the communication module and tutorial groups overview. This feature should now be extended to the user lists when managing a course and for server administration.

### Description

I added a new component to make it easier to use the profile pictures on the platform. I changed the current code to the component to the communication messages, communication user list, tutorial group overview. Then I added the component to the user administration list and course management lists.

### Steps for Testing

Prerequisites:
- 1 Instructor or Admin

1. Log in to Artemis
2. Navigate to course management
3. On the top right of a course card click the corresponding user group to get to the list.
4. Check if the profile pictures are displayed correctly.
5. As the admin, go to "Server Administration" > "User Management".
6. Check if in the user list the profile pictures are also displayed correctly.

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| tutorial-group-detail.component.ts | 96.22% | ✅  |
| conversation-member-row.component.ts | 84.03% | ✅ |
| posting-header.directive.ts | 100% | ✅ |
| profile-picture.component.ts | 100% | ✅ |

### Screenshots

Bildschirmfoto 2024-10-21 um 16 49 08

Bildschirmfoto 2024-10-21 um 16 49 22

## Release Notes

- **New Features**
  - Introduced `ProfilePictureComponent` for displaying user profile pictures across various modules.
  - Enhanced user management and course group interfaces with new columns for profile pictures.
  - Added localization support for profile picture labels in both English and German.

- **Bug Fixes**
  - Streamlined handling of profile picture rendering, improving performance and maintainability.

- **Tests**
  - Updated test suites to include `ProfilePictureComponent`, ensuring robust testing of profile picture functionalities.

### CODE REVIEW

**Review by sachmii — APPROVED** [review]
Tested on TS2 as a instructor, works fine. Nice feature :)

File: src/main/webapp/i18n/de/user-management.json [hunk 1]
```diff
@@ -102,7 +102,8 @@
             "search": "Suche",
             "loading": "Laden...",
             "searchError": "Der Suchbegriff muss leer oder mindestens 3 Zeichen lang sein.",
-            "jenkinsChange": "Die Änderung des Benutzernamens wird für dein aktuelles Setup (Jenkins) nicht unterstützt. Möglicherweise musst du den Benutzernamen in Jenkins manuell ändern.\nDu hast den Benutzer <i>{{ oldLogin }}</i> auf <i>{{ newLogin }}</i> aktualisiert."
+            "jenkinsChange": "Die Änderung des Benutzernamens wird für dein aktuelles Setup (Jenkins) nicht unterstützt. Möglicherweise musst du den Benutzernamen in Jenkins manuell ändern.\nDu hast den Benutzer <i>{{ oldLogin }}</i> auf <i>{{ newLogin }}</i> aktualisiert.",
```
Comment (ole-ve) [inline src/main/webapp/i18n/de/user-management.json:105]: Out of curiosity regarding `Du hast den Benutzer <i>{{ oldLogin }}</i> auf <i>{{ newLogin }}</i> aktualisiert.`.

Is it the Artemis user that has been changed? I'd probably add that info.

**Review by ole-ve — APPROVED** [review]
Code looks good, just one smaller suggestion

File: src/main/webapp/i18n/de/user-management.json (hunk 1 above)
Comment (PaRangger) [reply src/main/webapp/i18n/de/user-management.json:105]: I am actually unsure which user is meant there 😅 I did not add this line of text, i just added a comma to that line to add another string.

**Review by flbrgit — COMMENTED** [review]
Tested on TS2 as instructor. Profile picture in group list correctly rendered without problems
Screenshot 2024-10-22 185851

**Review by SindiBuklaji — DISMISSED** [review]
Tested on TS3. It shows the profile pictures as described 👍
image

**Review by kevinfischer4 — COMMENTED** [review]
Tested on TS5 as Instructor. Profile picture is shown as expected.
Screenshot 2024-10-22 211207

File: src/main/webapp/app/shared/profile-picture/profile-picture.component.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+@Component({
+    selector: 'jhi-profile-picture',
+    standalone: true,
+    templateUrl: './profile-picture.component.html',
+    styleUrls: ['./profile-picture.component.scss'],
+    imports: [ArtemisSharedCommonModule, RouterLink, ArtemisSharedPipesModule],
+})
+export class ProfilePictureComponent implements OnInit, OnChanges {
+    readonly imageSizeInRem = input<string>('2.15');
+    readonly fontSizeInRem = input<string>('0.9');
```
Comment (JohannesStoehr) [inline src/main/webapp/app/shared/profile-picture/profile-picture.component.ts:18]: As far as I can tell they always have the same ratio for all inputs. Can we only have one as input and calculate the other?

File: src/main/webapp/app/shared/profile-picture/profile-picture.component.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    readonly defaultPictureId = input<string>('');
+    readonly isEditable = input<boolean>(false);
+
+    profilePictureBackgroundColor: string;
+    userProfilePictureInitials: string;
+    imageSize: string;
+    fontSize: string;
+
+    // Icons
+    faCog = faCog;
```
Comment (JohannesStoehr) [inline src/main/webapp/app/shared/profile-picture/profile-picture.component.ts:34]:
```suggestion
    protected readonly faCog = faCog;
```

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]
Code looks good overall, left two nitpicks:

File: src/main/webapp/i18n/de/user-management.json (hunk 1 above)
Comment (ole-ve) [reply src/main/webapp/i18n/de/user-management.json:105]: It's updating the Artemis-User, but not the Jenkins-User. See UserService#updateUserInConnectorsAndAuthProvider.

**Review by HawKhiem — APPROVED** [review]
Tested on TS5. Profile pics shown as described!

File: src/main/webapp/app/shared/profile-picture/profile-picture.component.ts (hunk 2 above)
Comment (PaRangger) [reply src/main/webapp/app/shared/profile-picture/profile-picture.component.ts:18]: I was trying to do that but unfortunately the font size is not in a direct ratio with the imagesize for most things. Tutorial Group is 4.5:2 = 2.25, here it is 2.15:0.9 = 2.38, in the communication channel lists its 2:0.8=2.5, and so on.

File: src/main/webapp/app/shared/profile-picture/profile-picture.component.ts (hunk 3 above)
Comment (PaRangger) [reply src/main/webapp/app/shared/profile-picture/profile-picture.component.ts]: Thanks for the feedback! Updated in this [commit](https://github.com/ls1intum/Artemis/pull/9553/commits/2ea5314c1c9a2168367644fcf2f239d3ce2f9632).

**Review by krusche — APPROVED** [review]

**Pull request merged**